File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
       int max=0;
       int min=9999999;
    
    for(int i=0;i<prices.length;i++)
    {
        if(prices[i]<min)
        {
            min=prices[i];
        }
        int val=prices[i]-min;
        if(val>max)
        {
            max=val;
        }
    }
    return max;


        
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max=0
        min=999999999
        val=0
        for i in prices:
            if i<min:
                min=i
            val=i-min
            if val>max:
                max=val
        return max
        